src/popup/components/AddressBook/AddressBookList.vue

```diff
@@ -0,0 +1,144 @@
+<template>
+  <div class="address-book-list">
+    <IonHeader class="address-book-header">
+      <Transition name="fade-between">
+        <InputSearch
+          v-if="isScrolled || searchQuery"
+          v-model="searchQuery"
+          :placeholder="$t('pages.addressBook.searchPlaceholder')"
+          class="search-field"
+        />
+      </Transition>
+
+      <AddressBookFilters />
+    </IonHeader>
+    <IonContent ref="scrollWrapperEl" class="ion-padding ion-content-bg">
+      <div
+        v-if="Object.keys(addressBookFiltered).length"
+        class="list"
+      >
+        <AddressBookItem
+          v-for="(entry) in addressBookFiltered"
+          :key="entry.name"
+          :item="entry"
+        />
+      </div>
+      <p v-else class="message" v-text="noRecordsMessage" />
+    </IonContent>
+  </div>
+</template>
+
+<script lang="ts">
+import {
+  computed,
+  defineComponent,
+  onBeforeUnmount,
+  onMounted,
+  ref,
+} from 'vue';
+import { throttle } from 'lodash-es';
+import { IonHeader, IonContent } from '@ionic/vue';
+
+import { tg } from '@/popup/plugins/i18n';
+import { ComponentRef } from '@/types';
+import { ADDRESS_BOOK_FILTERS } from '@/constants';
+import { useAddressBook } from '@/composables';
+
+import InputSearch from '@/popup/components/InputSearch.vue';
+import AddressBookItem from '@/popup/components/AddressBook/AddressBookItem.vue';
+import AddressBookFilters from '@/popup/components/AddressBook/AddressBookFilters.vue';
+
+export default defineComponent({
+  components: {
+    IonHeader,
+    IonContent,
+    AddressBookItem,
+    AddressBookFilters,
+    InputSearch,
+  },
+  props: {
+    modelValue: Boolean,
+  },
+  emits: ['update:modelValue'],
+  setup(_, { emit }) {
+    const scrollWrapperEl = ref<ComponentRef>();
+    const isScrolled = ref(false);
+
+    const { addressBookFiltered, activeFilter, searchQuery } = useAddressBook();
+
+    const noRecordsMessage = computed(() => {
+      switch (true) {
+        case !!searchQuery.value: return tg('pages.addressBook.noRecords.search');
+        case activeFilter.value === ADDRESS_BOOK_FILTERS.all: return tg('pages.addressBook.noRecords.addressBook');
+        case activeFilter.value === ADDRESS_BOOK_FILTERS.bookmarked: return tg('pages.addressBook.noRecords.bookmarked');
+        default: return tg('pages.addressBook.noRecords.blockchain');
+      }
+    });
+    const scrollWrapper = computed(() => scrollWrapperEl.value?.$el);
+
+    function handleScroll() {
+      if (!scrollWrapper.value) return;
+      isScrolled.value = (scrollWrapper.value as HTMLElement)?.scrollTop > 0;
+      emit('update:modelValue', (isScrolled.value || !!searchQuery.value));
+    }
+    const throttledHandleScroll = throttle(handleScroll, 100);
+
+    onMounted(async () => {
+      scrollWrapper.value?.addEventListener('scroll', throttledHandleScroll);
+    });
+
+    onBeforeUnmount(() => {
```

Comment:
This will not trigger while you are on `Address book` page upon clicking on `Add` button or going to edit mode. Also it won't trigger if user will click close button in these `Add` and  `Edit` pages.